client/asset/btc/wallettypes.go

```diff
@@ -5,6 +5,21 @@ package btc
 
 import "decred.org/dcrdex/dex"
 
+// GetBalancesResult models a successful response from the getbalances request.
+type GetBalancesResult struct {
+	Mine struct {
+		Trusted   float64  `json:"trusted"`
+		Untrusted float64  `json:"untrusted_pending"`
+		Immature  float64  `json:"immature"`
+		Used      *float64 `json:"used,omitempty"`
```

Comment:
```
"trusted" : n,              (numeric) trusted balance (outputs created by the wallet or confirmed outputs)
"untrusted_pending" : n,    (numeric) untrusted pending balance (outputs created by others that are in the mempool)
"immature" : n,             (numeric) balance from immature coinbase outputs
"used" : n                  (numeric) (only present if avoid_reuse is set) balance from coins sent to addresses that were previously spent from (potentially privacy violating)
```

So `"trusted"` includes 0-conf outputs created by the wallet and confirmed outputs, while `"untrusted_pending"` can be thrown in the same bin as `"immature"` for the purposes of DEX wallet balance.

Reply:
Bumps our `minNetworkVersion` to `190000`, and we lose litecoin which isn't there yet.

I was on 0.18. After building 0.20 (released a couple of weeks ago), I was seeing a `Fee estimation failed. Fallbackfee is disabled` error. I had to set `-fallbackfee=0.00001` in the calls to `bitcoind` in the harness.

Reply:
Any chance you're running Bitcoin with `blocksonly`?  https://bitcoin.stackexchange.com/q/88621

I'm more than fine waiting for LTC.

Reply:
Oh, https://github.com/bitcoin/bitcoin/pull/16524

Reply:
It seems that Bitcoin Core uses a combination of fees in past blocks and mempool history to estimate fees: https://github.com/bitcoin/bitcoin/blob/35ed88f187c9bc7eedc3a1cf12193e0fbf222057/src/policy/fees.h#L120-L125

So being fully syncd apparently is not enough to allow smart fee estimation to work.  The node has to get "sufficient data" from mempool too to estimate fees, in the words of the `-fallbackfee` docs:

```
  -fallbackfee=<amt>
       A fee rate (in BTC/kB) that will be used when fee estimation has
       insufficient data. 0 to entirely disable the fallbackfee feature.
       (default: 0.00)
```

I wish I knew what "sufficient data" was (some likely outdata description [here](https://gist.github.com/morcos/d3637f015bc4e607e1fd10d8351e9f41)), and that there was a way to query if it has this before trying to make transactions.

But it's interesting that we would have hit this when going to mainnet if it were not for this 0.20 change that disabled the fallback fee for testnet and regnet.

Reply:
> But it's interesting that we would have hit this when going to mainnet

Probably not though, since mainnet probably always has "sufficient data".

Reply:
> I'm more than fine waiting for LTC.

If you mean drop LTC support until they get up to 0.19, then I'm definitely not on board. Who knows when that'll be, and the old algorithm can be adapted in the meantime. I see a couple of options, both relying on a `LegacyBalance` function. 

1. We could switch on RPC version, using `LegacyBalance` if < 0.19. 
2. A new type, `ltc.ExchangeWallet`, with embeds `btc.ExchangeWallet` and does something like
```
type ExchangeWallet struct {
	btc.ExchangeWallet
}

func (w *ExchangeWallet) Balance() (*asset.Balance, error) {
	return w.LegacyBalance()
}
```

If, on the other hand, you just meant that you can wait until I fix it in a follow up PR, then that's fine with me.

Reply:
Trolling LTC for 0.19, not that they care: https://twitter.com/chappjc/status/1273405424419721217

Reply:
> > But it's interesting that we would have hit this when going to mainnet
> 
> Probably not though, since mainnet probably always has "sufficient data".

Well it seems that this can mean a warm up period for your node to get mempool data to gauge congestion, hence why [`blocksonly=1` disables fee estimation](https://github.com/bitcoin/bitcoin/blob/fa8ced32a60dea37ac169241cf9a1f708ef46c4b/doc/reduce-traffic.md#4-turn-off-transaction-relay--blocksonly).  The following Reddit responses are vague and uninformative, but generally corroborated by other sources:  https://www.reddit.com/r/Bitcoin/comments/dnw5h3/why_this_warning_fee_estimation_is_currently_not/

> If you mean drop LTC support until they get up to 0.19, then I'm definitely not on board

That is actually what I meant, but we can work around it in the way you suggested in a follow up PR.  I just don't see a pressing need for LTC support in DEX MVP as we'll have our hands full supporting DEX on the DCR-BTC pair for months, and I expect LTC is feeling a bit of pressure to catch up with BTC (they rolled the LTC 0.18 release exactly 8 days after BTC's 0.20 release, out of embarrassment I suspect).

Reply:
There does seem to be a 0.20 branch of litecoin in their dev's fork.  I've made an issue for us to evaluate it: https://github.com/decred/dcrdex/issues/508